File: 0001-two-sum/0001-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numsMap: [Int: Int] = [:]
        
        for (i, num) in nums.enumerated() {
            var complement = target - num
            
            if let foundComplement = numsMap[complement] {
                return [foundComplement, i]
            }
            numsMap[num] = i
        }
        
        return []
    }
}

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    hash := make(map[int]int)
    
    for i, num := range nums {
        complement := target - num
        
        if _, ok := hash[complement]; ok {
            return []int{hash[complement], i}
        }
        
        hash[num] = i
    }
    
    return []int{-1, -1}
}